python/ray/data/dataset.py

```diff
@@ -1285,7 +1442,13 @@ def mean(
             If the dataset is empty, all values are null, or any value is null
             AND ``ignore_nulls`` is ``False``, then the output will be None.
         """
-        ret = self._aggregate_on(Mean, on, ignore_nulls)
+        dataset_format = self._dataset_format()
+        if dataset_format in ("arrow", "pandas"):
+            # Tabular data fast path.
+            ret = self._aggregate_vectorized_on(VectorizedMean, on, ignore_nulls)
+        else:
+            # Fall back to groupby + accumulator implementation for simple datasets.
+            ret = self._aggregate_on(Mean, on, ignore_nulls)
```

Comment:
Let's try to push the vectorization down as deep as possible, so it only affects the leaf task nodes (e.g., combine() function)